charm/charmcraft.yaml

```diff
@@ -0,0 +1,53 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+name: httprequest-lego-provider
+type: charm
+
+bases:
+  - build-on:
+    - name: ubuntu
+      channel: "22.04"
+    run-on:
+    - name: ubuntu
+      channel: "22.04"
+
+summary: HTTPRequest Lego provider.
+
+description: |
+  A [Juju](https://juju.is/) [charm](https://juju.is/docs/olm/charmed-operators)
+  deploying and managing [HTTP request Lego provider](https://go-acme.github.io/lego/dns/httpreq/)
+  on Kubernetes. HTTP request Lego provider is a web application implementing the [DNS HTTP request
+  provider as defined by ACME](https://go-acme.github.io/lego/dns/httpreq/) to manage DNS records.
+
+  This charm simplifies initial deployment and "day N" operations of the HTTP request Lego
+  provider on Kubernetes, enabling the automation of DNS management. It allows for deployment
+  on many different Kubernetes platforms, from [MicroK8s](https://microk8s.io) to
+  [Charmed Kubernetes](https://ubuntu.com/kubernetes) and public cloud Kubernetes offerings.
+
+  As such, the charm makes it easy for those looking to host their own ACME provider, and gives
+  them the freedom to deploy on the Kubernetes platform of their choice.
+
+  For DevOps or SRE teams this charm will make operating HTTP request Lego provider simple and
+  straightforward through Juju's clean interface. It will allow easy deployment
+  into multiple environments for testing of changes, and supports scaling out for
+  enterprise deployments.
+
+extensions:
+  - django-framework
+
+requires:
+  postgresql:
+    interface: postgresql_client
+    limit: 1
+
+config:
+  options:
+    dns_repository_url:
+      type: string
+      description: The repository URL where the DNS records are stored.
+    secret_key:
+      type: string
+      description: Django secret key.
```

Comment:
Isn't secret_key automatically managed by the charm ?

Reply:
Not that I'm aware of. Is it @weiiwang01 ?

Reply:
Should be: https://github.com/canonical/xiilib/blob/feat-12f-django/xiilib/django/charm.py#L94

Reply:
Re-opening, as I think there's a misunderstanding
This is managed by the 12-factor, so unnecessary:
https://github.com/weiiwang01/charmcraft/blob/feat-12f-django/charmcraft/extensions/django.py#L40-L43

The extension will automatically add a config option for it, and the charm will inject it into your workload as "DJANGO_SECRET_KEY" environment variable

Reply:
Ahh. Modified